models/issue.go

```diff
@@ -1193,37 +1193,43 @@ type IssuesOptions struct {
 func sortIssuesSession(sess *xorm.Session, sortType string, priorityRepoID int64) {
 	switch sortType {
 	case "oldest":
-		sess.Asc("issue.created_unix")
+		sess.Asc("issue.created_unix").Asc("issue.id")
 	case "recentupdate":
-		sess.Desc("issue.updated_unix")
+		sess.Desc("issue.updated_unix").Desc("issue.id")
 	case "leastupdate":
-		sess.Asc("issue.updated_unix")
+		sess.Asc("issue.updated_unix").Asc("issue.id")
 	case "mostcomment":
-		sess.Desc("issue.num_comments")
+		sess.Desc("issue.num_comments").Desc("issue.id")
 	case "leastcomment":
-		sess.Asc("issue.num_comments")
+		sess.Asc("issue.num_comments").Desc("issue.id")
```

Comment:
```suggestion
		sess.Asc("issue.num_comments").Asc("issue.id")
```

Reply:
The rationale here is that when sorting by the number of comments the order of creation do not matter, so in both cases fallback to default, i.e. descending order, i.e. newest first.

Reply:
Shouldn't it be `sess.Asc("issue.num_comments").Desc("issue.created_unix").Desc("issue.id")` in that case?

Reply:
Make sense. Addressed in newest commit.